ArcGISRuntimeSDKQt_CppSamples/Analysis/TraceUtilityNetwork/TraceUtilityNetwork.cpp

```diff
@@ -164,38 +184,66 @@ void FindFeaturesUtilityNetwork::multiTerminalIndex(int index)
   updateTraceParams(element);
 }
 
-void FindFeaturesUtilityNetwork::updateTraceParams(UtilityElement* element)
+void TraceUtilityNetwork::updateTraceParams(UtilityElement* element)
 {
   if (m_startingLocationsEnabled)
   {
     m_startingLocations.append(element);
-    m_traceParams->setStartingLocations(m_startingLocations);
     Graphic* traceLocation = new Graphic(m_clickPoint, m_startingSymbol, m_graphicParent.get());
     m_graphicsOverlay->graphics()->append(traceLocation);
   }
   else
   {
     m_barriers.append(element);
-    m_traceParams->setBarriers(m_barriers);
     Graphic* traceLocation = new Graphic(m_clickPoint, m_barrierSymbol, m_graphicParent.get());
     m_graphicsOverlay->graphics()->append(traceLocation);
   }
 }
 
-void FindFeaturesUtilityNetwork::trace()
+void TraceUtilityNetwork::trace(int index)
 {
   m_busy = true;
   emit busyChanged();
+
+  if (m_traceParams)
+    delete m_traceParams;
```

Comment:
You can remove this check. Deleting nullptr is well-defined behavior.

https://isocpp.org/wiki/faq/freestore-mgmt#delete-handles-null